src/Razor/src/Microsoft.CodeAnalysis.Razor/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -63,13 +63,20 @@ public void Execute(TagHelperDescriptorProviderContext context)
             for (var i = 0; i < types.Count; i++)
             {
                 var type = types[i];
-                var descriptor = CreateDescriptor(symbols, type);
-                context.Results.Add(descriptor);
 
-                foreach (var childContent in descriptor.GetChildContentProperties())
+                // Components have very simple matching rules.
+                // 1. The type name (short) matches the tag name.
+                // 2. The fully qualified name matches the tag name.
+                var shortNameMatchingDescriptor = CreateShortNameMatchingDescriptor(symbols, type);
+                context.Results.Add(shortNameMatchingDescriptor);
+                var fullyQualifiedNameMatchingDescriptor = CreateFullyQualifiedNameMatchingDescriptor(symbols, type);
```

Comment:
Do you plan on having two descriptors as the long term plan or is this just for now?

Reply:
I haven't decided yet. It will depend on how I support using alias (It is complicated because I'll need to change the tag matching rule based on what alias was parsed for a specific namespace or type). It is going to need a lot more thought.